ui-participant/src/providers/I18nProvider.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { createContext, useContext, useEffect, useState } from 'react'
+import { LanguageText } from '@juniper/ui-core'
+import Api from 'api/api'
+import { useUser } from './UserProvider'
+
+const I18nContext = createContext<I18nContextT | null>(null)
+
+export type I18nContextT = {
+  languageTexts: LanguageText[]
```

Comment:
nit: transform this to a map? I could see the `i18n` function being called many times in one page and making it a map would make it a constant-time operation.

I'm also _slightly_ worried about the fact that the languageTexts could get very big - we could maybe have sandboxed `i18n` providers, like e.g. `<I18nProvider area='homepage'>` which will grab just the translations for the homepage, etc. for each area of the application. But that is likely total overengineering and we don't need to do that. Just a thought - I think it's in a great place right now.

Reply:
I totally agree, I initially shared that same worry and started down that path in the original implementation, but yeah it ended up feeling like overkill for where we're at right now. It's definitely something to keep an eye on though as app functionality expands over time

Reply:
agree we don't need section-specific modules yet.  What might be easy to do though is to have the language texts be sent from the server to the client already as a map.  check out the `json_object_agg` function of postgresql -- you might be able to make a DAO method that just returns the texts already in a json hash by key.  Alternatively, you could just put them in a hash in Java, and then use Spring caching to cache the hash